From 56a857a713f49d867cedd02eca5004202e9d9a5c Mon Sep 17 00:00:00 2001
From: Zidan Wang <zidan.wang@freescale.com>
Date: Fri, 27 Nov 2015 11:14:36 +0800
Subject: [PATCH 0701/5242] MLK-11915-05 ASoC: imx-wm8958: init codec_np to
 avoid wild pointer

commit  6d8e6cd5e1de8270fba2ea355a02836530b1994a from
https://source.codeaurora.org/external/imx/linux-imx.git

init codec_np to avoid wild pointer. Reported by Coverity.

Signed-off-by: Zidan Wang <zidan.wang@freescale.com>
(cherry picked from commit 76665930654867cf38a86ba747a9f8a5bf2665e2)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-wm8958.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/fsl/imx-wm8958.c b/sound/soc/fsl/imx-wm8958.c
index 74486f0..047bdf9 100644
--- a/sound/soc/fsl/imx-wm8958.c
+++ b/sound/soc/fsl/imx-wm8958.c
@@ -403,7 +403,7 @@ static int imx_wm8958_set_bias_level_post(struct snd_soc_card *card,
 
 static int imx_wm8958_probe(struct platform_device *pdev)
 {
-	struct device_node *cpu_np, *codec_np, *gpr_np;
+	struct device_node *cpu_np, *codec_np = NULL, *gpr_np;
 	struct device_node *np = pdev->dev.of_node;
 	struct platform_device *cpu_pdev;
 	struct imx_priv *priv = &card_priv;
-- 
1.7.9.5

